lxd/resources/cpu.go

```diff
@@ -227,7 +232,68 @@ func GetCPU() (*api.ResourcesCPU, error) {
 	}
 
 	defer func() { _ = f.Close() }()
-	cpuInfo := bufio.NewScanner(f)
+	cpuInfoScanner := bufio.NewScanner(f)
+	cpuInfoMap := map[int64]*cpuInfo{}
+
+	// CPU information
+	for cpuInfoScanner.Scan() {
+		line := strings.TrimSpace(cpuInfoScanner.Text())
+		if !strings.HasPrefix(line, "processor") {
+			return nil, fmt.Errorf("Failed to parse /proc/cpuinfo: Unexpected line %q", line)
+		}
+
+		// Extract cpu index
+		fields := strings.SplitN(line, ":", 2)
```

Comment:
strings.Cut would likely be more appropriate here as an FYI only

Reply:
https://github.com/canonical/lxd/pull/14683/commits/de6416c7f889b735592a0d47b5b7655979a0b757 is my attempt at using `strings.Cut`.